### PR TITLE
Dev background task ios syncing

### PR DESCRIPTION
**Pull request by juanky201271**
Tasks:
* IOS background syncing setting up.
* Added AppState to control background/foreground responsively.
* Upgraded the react-native version.
* using the new JS Engine, Hermes.
* New feature `send All` settings & button.

### CODE REVIEW

File: app/LoadedApp/LoadedApp.tsx [hunk 1]
```diff
@@ -213,9 +248,11 @@ class LoadedAppClass extends Component<LoadedAppClassProps, AppStateLoaded> {
       this.setInfo,
       this.refreshUpdates,
       props.translate,
+      this.fetchBackgroundSyncing,
     );
 
     this.dim = {} as EmitterSubscription;
+    this.sta = {} as NativeEventSubscription;
```
Comment (AloeareV) [inline app/LoadedApp/LoadedApp.tsx]:
```suggestion
    this.state = {} as NativeEventSubscription;
```

File: app/LoadedApp/LoadedApp.tsx [hunk 2]
```diff
@@ -228,11 +265,31 @@ class LoadedAppClass extends Component<LoadedAppClassProps, AppStateLoaded> {
       this.setDimensions(screen);
       //console.log('++++++++++++++++++++++++++++++++++ change dims', Dimensions.get('screen'));
     });
+
+    this.sta = AppState.addEventListener('change', async nextAppState => {
```
Comment (AloeareV) [inline app/LoadedApp/LoadedApp.tsx]:
```suggestion
    this.state = AppState.addEventListener('change', async nextAppState => {
```

File: app/LoadedApp/LoadedApp.tsx [hunk 3]
```diff
@@ -228,11 +265,31 @@ class LoadedAppClass extends Component<LoadedAppClassProps, AppStateLoaded> {
...
+      }
+      this.setState({ appState: nextAppState });
+    });
   };
 
   componentWillUnmount = () => {
     this.rpc.clearTimers();
-    this.dim?.remove();
+    this.dim.remove();
+    this.sta.remove();
```
Comment (AloeareV) [inline app/LoadedApp/LoadedApp.tsx]:
```suggestion
    this.state.remove();
```

File: app/LoadedApp/LoadedApp.tsx [hunk 4]
```diff
@@ -175,11 +204,14 @@ type LoadedAppClassProps = {
...
   currency: 'USD' | '';
   server: string;
+  sendAll: boolean;
+  background: backgroundType;
 };
 
 class LoadedAppClass extends Component<LoadedAppClassProps, AppStateLoaded> {
   rpc: RPC;
   dim: EmitterSubscription;
+  sta: NativeEventSubscription;
```
Comment (AloeareV) [inline app/LoadedApp/LoadedApp.tsx]: Not a blocking concern: If I had to guess, sta is "state" but it's not immediately obvious to me. Could we call it 'state'?
```suggestion
  state: NativeEventSubscription;
```

File: app/LoadingApp/LoadingApp.tsx [hunk 5]
```diff
@@ -221,10 +246,28 @@ class LoadingAppClass extends Component<LoadingAppClassProps, AppStateLoading> {
...
+        console.log('App is gone to the background!');
+      }
+      this.setState({ appState: nextAppState });
+    });
   };
 
   componentWillUnmount = () => {
-    this.dim?.remove();
+    this.dim.remove();
+    this.sta.remove();
```
Comment (AloeareV) [inline app/LoadingApp/LoadingApp.tsx]:
```suggestion
    this.state.remove();
```

**Review by AloeareV — DISMISSED** [review]

**Comment by juanky201271** [comment]
@AloeareV I did the change you suggested, but I changed sta -> appstate, because the `state` class property already exists.

**Review by AloeareV — APPROVED** [review]

**Pull request merged**